vcf.c

```diff
@@ -1236,9 +1236,9 @@ static inline int bcf_read1_core(BGZF *fp, bcf1_t *v)
     shared_len = le_to_u32(x);
     if (shared_len < 24) return -2;
     shared_len -= 24; // to exclude six 32-bit integers
-    if (ks_resize(&v->shared, shared_len) != 0) return -2;
+    if (ks_resize(&v->shared, shared_len ? shared_len : 1) != 0) return -2;
     indiv_len = le_to_u32(x + 4);
-    if (ks_resize(&v->indiv, indiv_len) != 0) return -2;
+    if (ks_resize(&v->indiv, indiv_len ? indiv_len : 1) != 0) return -2;
```

Comment:
Isn't this a generic issue of ks_resize with size 0, rather than a specific VCF issue?

If so it'd be better fixed in kstring.h

Reply:
Everything `ks_resize()` does is wrapped in `if (s->m < size)` which is never true for `size == 0`.  I think it would be more intrusive to try to change this behaviour instead of just fixing the one or two places where it may cause a problem.